### PR TITLE
Citrix CI: fix logs upload failure

### PR DESCRIPTION
**Pull request by naichuans**
Because of openstack sdk upgrade, connection create would be failed
because of API changes. Used new API to fix this error.
Need install pkg os_client_config and set cloud.yaml.

### CODE REVIEW

File: swiftuploader/upload.py [hunk 1]
```diff
@@ -109,28 +114,24 @@ def sizeof_fmt(num, suffix='B'):
...
-        username=username,
-        password=password
-    )
+def create_connection(password):
+    cloud_conf = open(CLOUD_CONF,"r+")
+    s=cloud_conf.read()
+    cloud_conf.seek(0,0)
+    cloud_conf.write(s.replace("XXXXXX", "\'" + password + "\'"))
+    cloud_conf.close()
+    opts = Opts(cloud_name="mordred")
```
Comment (jianghuaw) [inline swiftuploader/upload.py]: what's the meaning of mordred? I guess we need a more meaningful name: e.g. rax?
Also wonder if we should avoid hard coding cloud_name.

File: swiftuploader/upload.py [hunk 2]
```diff
@@ -109,28 +114,24 @@ def sizeof_fmt(num, suffix='B'):
...
-        project_name=project_name,
-        username=username,
-        password=password
-    )
+def create_connection(password):
+    cloud_conf = open(CLOUD_CONF,"r+")
+    s=cloud_conf.read()
+    cloud_conf.seek(0,0)
+    cloud_conf.write(s.replace("XXXXXX", "\'" + password + "\'"))
+    cloud_conf.close()
```
Comment (jianghuaw) [inline swiftuploader/upload.py]: Should we move these lines update configure file into a separate function?

File: swiftuploader/upload.py [hunk 3]
```diff
@@ -12,27 +13,31 @@
 from openstack import profile
 from openstack import utils
 
+CLOUD_CONF = "/home/jenkins/.config/openstack/clouds.yaml"
```
Comment (jianghuaw) [inline swiftuploader/upload.py]: We should avoid depending on this too specific path or user - jenkins. May input the path via command line args.

File: clouds.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+clouds:
+ mordred:
```
Comment (jianghuaw) [inline clouds.yaml]: mordred?

File: swiftuploader/upload.py [hunk 5]
```diff
@@ -12,27 +13,31 @@
...
+
+
 class UploadException(Exception):
     pass
 
+
+class Opts(object):
```
Comment (jianghuaw) [inline swiftuploader/upload.py]: Is "Opts" a too general name?

File: swiftuploader/upload.py [hunk 6]
```diff
@@ -12,27 +13,31 @@
...
+    def __init__(self, cloud_name='devstack-admin', debug=False):
+        self.cloud = cloud_name
+        self.debug = debug
+        # Use identity v3 API for examples.
+        self.identity_api_version = '3'
```
Comment (jianghuaw) [inline swiftuploader/upload.py]: can we make this an input args for __init__ and use '3' as default value?

**Review by jianghuaw — CHANGES_REQUESTED** [review]

File: swiftuploader/upload.py [hunk 7]
```diff
@@ -3,36 +3,44 @@
...
 import time
 import hashlib
+import yaml
 
 from openstack import connection
-from openstack import profile
-from openstack import utils
+
+
+PWD_COVER = 'XXXXXX'
```
Comment (BobBall) [inline swiftuploader/upload.py]: Do you mean that this is just a replaced string?  If so this should be "<password>" or similar

File: swiftuploader/upload.py [hunk 8]
```diff
@@ -108,29 +116,32 @@ def sizeof_fmt(num, suffix='B'):
...
+        yaml.dump(clouds_dict, f)
+
+def create_connection(password, cloud_name):
+    set_cloud_password(password, cloud_name)
+    opts = cloud_conf(cloud_name)
+    occ = os_client_config.OpenStackConfig()
+    cloud = occ.get_one_cloud(opts.cloud)
+    conn = connection.from_config(cloud_config=cloud, options=opts)
+    set_cloud_password(PWD_COVER, cloud_name)
     conn.authorize()
```
Comment (jianghuaw) [inline swiftuploader/upload.py:137]: confused why this is needed.

File: swiftuploader/upload.py (hunk 7 above)
Comment (naichuans) [reply swiftuploader/upload.py]: Yes, it just used to put in clouds.yaml as a place holder. Will change to ""

File: swiftuploader/upload.py (hunk 8 above)
Comment (naichuans) [reply swiftuploader/upload.py:137]: Just want to prevent the password to be seen, at most of the times...

File: requirements.txt [hunk 9]
```diff
@@ -1,2 +1,3 @@
 pbr<1.0.0
 python-openstacksdk
+os_client_config
```
Comment (jianghuaw) [inline requirements.txt]: I guess the package name should be os-client-config?
Also please sort the package names.

**Review by jianghuaw — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 9 above)
Comment (naichuans) [reply requirements.txt]: Both work. Will modify it

**Pull request merged**